docs/tracer/README.md

```diff
@@ -0,0 +1,39 @@
+# Cairo Tracer
+
+Cairo-vm offers a tracer which gives you a visualization of how your memory and registers change line after line as the VM executes the code.
+
+## Setting up the tracer
+
+To use the tracer, you need to build your the VM using the `with_tracer` feature flag.
+
+```bash
+cargo build --release --features with_tracer
+```
+
+## Running the tracer
+
+Once the build in the above step is complete, you can use the `cairo-vm-cli` as you do normally with an additional argument `--tracer true`. This tells the VM to start a server where you can visualize the exection of your cairo program.
+
+```bash
+target/release/cairo-vm-cli <path_to_compiled_cairo_json> --layout <layout> --memory_file <path_to_store_memory_binary> --trace_file <path_to_store_tracer_binary> --tracer true
+```
+
+### NOTE
+> The `--memory_file` and `--trace_file` arguments are compulsary at the moment as the current code relocates the memory and trace only when these arguments are supplied. However, edits can be made in future versions to relocate if only the `--tracer` option is specified.
```

Comment:
```suggestion
> The `--memory_file` and `--trace_file` arguments are compulsory at the moment as the current code relocates the memory and trace only when these arguments are supplied. However, edits can be made in future versions to relocate if only the `--tracer` option is specified.
```
Typo

Reply:
Also, this requirement can be removed by setting the `trace_enabled` & `relocate_mem` flags to true when creating the `CairoRunConfig` in the `run` function of the vm's cli crate.